Scripts/Helpers/ProxyNode.cs

```diff
@@ -0,0 +1,626 @@
+                }
+
+                var renderer = affordance.renderer;
+                if (affordanceRenderers.Add(renderer))
+                {
+                    MaterialUtils.CloneMaterials(renderer); // Clone all materials associated with each renderer once
+
+                    // Clones that utilize the standard shader can lose their enabled ZWrite value (1), if it was enabled on the material
+                    foreach (var material in renderer.sharedMaterials)
+                    {
+                        material.SetFloat(k_ZWritePropertyName, 1);
+                    }
+                }
+
+                affordanceData.AddAffordance(renderer, affordance.tooltips);
+            }
+
+            foreach (var renderer in affordanceRenderers)
+            {
+                renderer.AddMaterial(m_ProxyBackgroundMaterial);
+            }
+
+            var bodyRenderers = GetComponentsInChildren<Renderer>(true)
+                .Where(x => !affordanceRenderers.Contains(x) && !IsChildOfProxyOrigin(x.transform)).ToList();
+
+            var bodyAffordanceDefinition = new AffordanceDefinition
+            {
+                visibilityDefinition = m_AffordanceMap.bodyVisibilityDefinition
+            };
+
+            m_BodyData = new AffordanceData(bodyAffordanceDefinition, this);
+            foreach (var renderer in bodyRenderers)
+            {
+                MaterialUtils.CloneMaterials(renderer);
+                m_BodyData.AddAffordance(renderer, null);
+                renderer.AddMaterial(m_ProxyBackgroundMaterial);
+            }
+        }
+
+        void Start()
+        {
+            if (m_ProxyAnimator)
+            {
+                m_ProxyAnimator.Setup(m_AffordanceData.Select(data => data.Value.definition).ToArray(), m_Affordances);
+                this.ConnectInterfaces(m_ProxyAnimator, rayOrigin);
+            }
+        }
+
+        void Update()
+        {
+            var cameraPosition = CameraUtils.GetMainCamera().transform.position;
+            var direction = GetFacingDirection(cameraPosition);
+            if (m_FacingDirection != direction)
+            {
+                m_FacingDirection = direction;
+                UpdateFacingDirection(direction);
+            }
+
+            var fadeInAmount = m_FadeInSpeedScalar * Time.deltaTime;
+            var fadeOutAmount = m_FadeOutSpeedScalar * Time.deltaTime;
+
+            foreach (var kvp in m_AffordanceData)
+            {
+                kvp.Value.Update(fadeInAmount, fadeOutAmount);
+            }
+
+            m_BodyData.Update(fadeInAmount, fadeOutAmount);
+        }
+
+        void OnDestroy()
+        {
+            StopAllCoroutines();
+
+            foreach (var kvp in m_AffordanceData)
+            {
+                kvp.Value.OnDestroy();
+            }
+
+            m_BodyData.OnDestroy();
+        }
+
+        FacingDirection GetFacingDirection(Vector3 cameraPosition)
+        {
+            var transform = m_NaturalOrientation;
```

Comment:
Why cache off m_NaturalOrientation, as opposed to just using m_NaturalOrientation?

Reply:
Ah, good catch. Laziness--I was using the builtin transform before. I'll clean this up :)